web/yo/app/scripts/controllers/drugs.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+angular.module('oncokbApp')
+    .controller('DrugsCtrl', ['$window', '$scope', '$rootScope', '$location', '$timeout', '$routeParams', '_', 'DTColumnDefBuilder', 'DTOptionsBuilder', 'DatabaseConnector', 'OncoKB', 'S', 'mainUtils', 'UUIDjs', '$q',
+        function($window, $scope, $rootScope, $location, $timeout, $routeParams, _, DTColumnDefBuilder, DTOptionsBuilder, DatabaseConnector) {
+
+            function getDrugList() {
+                DatabaseConnector.getAllDrugs().then(function(result){
+                    $scope.drugList = result;
+                })
+            };
+            getDrugList();
+
+            $scope.processSearchDrugs = function(keyword) {
+                return DatabaseConnector.searchDrugs(keyword).then(function(result) {
```

Comment:
Let's handle the failure here. What are you going to do if the API is not available?